lib/jsrt/JsrtSourceHolder.cpp

```diff
@@ -49,7 +51,6 @@ namespace Js
             *utf8Script = HeapNewArray(utf8char_t, cbUtf8Buffer);
         }
 
-        Assert(length < MAXLONG);
```

Comment:
I was speaking in general terms. Invariants are good to have (as documented by asserts). When they break, it's easier to find in debug builds. In release builds, the Asserts are removed by the pre-compiler.

However, if the if statement you introduce properly guards the bound, then there's no need for the assert as it would be redundant.